src/pip/_internal/resolution/resolvelib/candidates.py

```diff
@@ -0,0 +1,114 @@
+from pip._vendor.packaging.utils import canonicalize_name
+
+from pip._internal.req.constructors import install_req_from_line
+from pip._internal.req.req_install import InstallRequirement
+from pip._internal.utils.typing import MYPY_CHECK_RUNNING
+
+from .base import Candidate
+
+if MYPY_CHECK_RUNNING:
+    from typing import Dict, Optional, Sequence
+
+    from pip._internal.models.link import Link
+    from pip._internal.operations.prepare import RequirementPreparer
+    from pip._internal.resolution.base import InstallRequirementProvider
+
+    from pip._vendor.packaging.version import _BaseVersion
+    from pip._vendor.pkg_resources import Distribution
+
+
+# Dummy to make lint pass
+_CANDIDATE_CACHE = {}  # type: Dict[Link, Candidate]
```

Comment:
Is this comment stale or are there some future plans to this global?

Reply:
Yes, it's a stale comment. Will fix.